cirq/google/op_serializer.py

```diff
@@ -66,31 +66,44 @@ class GateOpSerializer:
         serialized_gate_id: The id used when serializing the gate.
     """
 
-    def __init__(self, *, gate_type: Type[Gate], serialized_gate_id: str,
-                 args: List[SerializingArg]):
+    def __init__(self,
+                 *,
+                 gate_type: Type[Gate],
+                 serialized_gate_id: str,
+                 args: List[SerializingArg],
+                 gate_predicate: Callable[[ops.Gate], bool] = None):
         """Construct the serializer.
 
         Args:
             gate_type: The type of the gate that is being serialized.
             serialized_gate_id: The string id of the gate when serialized.
+            gate_predicate: Sometimes a gate can only be serialized for
+                particular gate parameters. If this is not None, then
+                this predicate will be checked before attempting
+                to serialize the gate. If the predicate is False,
+                serialization will result in a None value.
             args: A list of specification of the arguments to the gate when
                 serializing, including how to get this information from the
                 gate of the given gate type.
         """
         self.gate_type = gate_type
         self.serialized_gate_id = serialized_gate_id
         self.args = args
+        self.gate_predicate = gate_predicate
 
-    def to_proto_dict(self, op: ops.GateOperation) -> Dict:
-        return json_format.MessageToDict(self.to_proto(op),
+    def to_proto_dict(self, op: ops.GateOperation) -> Optional[Dict]:
```

Comment:
I would rename the method to make it clearer that it is not guaranteed to return a serialization.

Reply:
Not a fan of "maybe" methods, especially now with type declarations.  Suggestion?